pontoon/sync/tasks.py

```diff
@@ -246,6 +234,20 @@ def sync_translations(
         )
         obsolete_vcs_entities = project_changes['obsolete_db']
 
+    # If project repositories have API access, we can retrieve latest commit hashes and detect
+    # changed locales before the expensive VCS pull/clone operations. When performing full scan and
+    # when source strings have changed, we still need to sync all locales.
```

Comment:
Or do we? I think we do for a full scan, but for en-US changes, I think we don't. We used to when we sync'ed removed strings and templates, but as we're only syncing files with l10n activity today, we don't?